packages/sdk/src/react/_internal/transaction-machine/execute-transaction.ts

```diff
@@ -26,22 +26,39 @@ import {
 	type Step,
 	StepType,
 } from '../../../types';
-
-export enum TransactionState {
-	IDLE = 'IDLE',
-	SWITCH_CHAIN = 'SWITCH_CHAIN',
-	CHECKING_STEPS = 'CHECKING_STEPS',
-	TOKEN_APPROVAL = 'TOKEN_APPROVAL',
-	EXECUTING_TRANSACTION = 'EXECUTING_TRANSACTION',
-	CONFIRMING = 'CONFIRMING',
-	SUCCESS = 'SUCCESS',
-	ERROR = 'ERROR',
+import { useTransactionStatusModal } from '../../ui/modals/_internal/components/transactionStatusModal';
```

Comment:
I wold prefer if we added this as a prop, instead of importing it directly.
If we want to write tests for this, it's much easier if we use dependency injection